### PR TITLE
Rewrite Profiler from scratch

### PR DESCRIPTION
**Pull request by marvinhagemeister**
Whilst debugging the underlying issue made visible in #359 and #343 I noticed that our internal data structures can be significantly simplified. The main source of complexity is that each stores `startTime` and `endTime` and the code uses that to detect if a node was rendered and additionally to lay down the flame graph layout. But this approach falls short when we're dealing with skipped subtrees like in #359 . I used some voodoo to align the timings of these nodes back into the main tree, but that's a pretty complicated machinery and way more code than I'd like.

So I was wondering what a better approach would look like that prevents these kind of problems from occurring in the first place. Upon sketching some ideas on paper I noticed that we can leverage our knowledge in the profiler of the tree's hierarchy. Armed with that we only care about how **long** a node rendered without excluding its children (=`selfDuration`). This allows us to just walk the tree and lay down the current node at the end of the parent. When we iterate over the children we simply pad all parents by the amount of `selfDuration` of the current node. And that's pretty much the whole algorithm and it's much simpler than what we had before.

As a nice benefit the flame graph layout has less "rivers" which always looked a bit odd and at times like a bug in the layout code.

**Before:**

Screenshot 2022-05-14 at 21 04 53

**After:**

Screenshot 2022-05-14 at 21 01 29

In summary this change in strategy **reduces the code needed for the profiler in half**. That's about 800 lines less than it was before. Overall I'm confident that this is a much better approach going forward and that it will squash a bunch of visual issues with the profiler.

Remaining tasks:

- [ ] Capture commit start time
- [ ] Scale past non-rendered sub trees to remaining place of parent
- [ ] Update unit tests
- [ ] Polishing
- [ ] Check memory consumption (should be less with this PR)

### CODE REVIEW

**Comment by marvinhagemeister** [comment]
Issue 1 to investigate:

Screenshot 2022-05-14 at 20 43 39
Screenshot 2022-05-14 at 20 43 44

**Comment by marvinhagemeister** [comment]
Issue 2 to investigate:

Screenshot 2022-05-14 at 20 44 47

**Comment by marvinhagemeister** [comment]
Issue: Search has stale results on reload (might be unrelated to profiler changes).

**Comment by marvinhagemeister** [comment]
Issue: Double `forwardRef`, really strange. Is probably unrelated to the Profiler too.

**Comment by marvinhagemeister** [comment]
Issue: Root filtering removes tree (probably a somewhat easy fix somewhere)

**Comment by marvinhagemeister** [comment]
Issue: Elements sidebar panel shows stale data on reloading.

**Comment by marvinhagemeister** [comment]
Issue: Error if previously selected node is not present in current commit. This is probably a glitch in the reactivity layer.
